bazel-mode.el

```diff
@@ -21,6 +21,35 @@
 ;;
 ;;; Code:
 
+(defgroup bazel-mode nil
+  "Major mode for Bazel BUILD files."
+  :link '(url-link "https://github.com/bazelbuild/emacs-bazel-mode")
+  :group 'languages)
+
+(defcustom bazel-mode-buildifier-cmd "buildifier"
+  "Filename of buildifier executable."
+  :type 'file
+  :group 'bazel-mode)
+
+(defun bazel-mode-buildifier ()
+  "Format current buffer using buildifier."
+  (interactive "*")
+  (let ((temp-file (make-temp-file "buildifier" nil nil (buffer-string)))
+        (input-buffer (current-buffer))
+        (buildifier-buffer (get-buffer-create "*buildifier*")))
+    (with-current-buffer buildifier-buffer
+      (erase-buffer)
+      (let ((return-code
+             (process-file buildifier-cmd temp-file buildifier-buffer nil "-type=build")))
+        (unwind-protect
+          (if (eq return-code 0)
+              (progn
+                (set-buffer input-buffer)
+                (replace-buffer-contents buildifier-buffer)
+                (kill-buffer buildifier-buffer))
+              (set-buffer-modified-p nil)
+              (display-buffer buildifier-buffer)))))))
```

Comment:
I'd use `with-temp-buffer-window` here, it has some nice behavior for such output buffers.

Reply:
Would you suggest using  with-temp-buffer above where I currently use with-current-buffer, and only creating the displayed *buildifier* buffer down here and display it using with-temp-buffer-window in case of an error code?

Reply:
Since `with-temp-buffer-window` displays the buffer unconditionally, you sould only call it in case of an error.

Reply:
Done.